templates/storage.template.yaml

```diff
@@ -164,23 +164,34 @@ Resources:
             Statement:
               - Effect: Allow
                 Action:
-                  - 'storagegateway:*'
-                Resource: '*'
+                  - 'storagegateway:ActivateGateway'
+                  - 'storagegateway:AddCache'
+                  - 'storagegateway:ListLocalDisks'
+                  - 'storagegateway:CreateNFSFileShare'
+                  - 'storagegateway:DeleteGateway'
+                  - 'storagegateway:DeleteFileShare'
+                  - 'storagegateway:AddTagsToResource'
+                Resource: !Sub 'arn:${AWS::Partition}:storagegateway:*:*:*'
+              - Effect: Allow
+                Action:
+                  - 'ec2:DescribeInstanceStatus'
+                  - 'ec2:CreateNetworkInterface'
+                  - 'ec2:DescribeNetworkInterfaces'
+                  - 'ec2:DeleteNetworkInterface'
+                  - 'ec2:AssignPrivateIpAddresses'
+                  - 'ec2:UnassignPrivateIpAddresses'
+                Resource: !Sub 'arn:${AWS::Partition}:ec2:*:*:*'
               - Effect: Allow
                 Action:
                   - 'ec2:DescribeInstanceStatus'
-                  - "ec2:CreateNetworkInterface"
-                  - "ec2:DescribeNetworkInterfaces"
-                  - "ec2:DeleteNetworkInterface"
-                  - "ec2:AssignPrivateIpAddresses"
-                  - "ec2:UnassignPrivateIpAddresses"
+                  - 'ec2:DescribeNetworkInterfaces'
                 Resource: '*'
               - Effect: Allow
                 Action:
                   - 'logs:CreateLogGroup'
                   - 'logs:CreateLogStream'
                   - 'logs:PutLogEvents'
-                Resource: '*'
+                Resource: !Sub 'arn:${AWS::Partition}:logs:*:*:*'
```

Comment:
Same as earlier. Can we provide specific resource name?

Reply:
Same as above. We don't create the Log group resource explicitly.